pkgs/os-specific/windows/mingw-w64/default.nix

```diff
@@ -1,22 +1,34 @@
 { lib, stdenv, windows, fetchurl }:
 
 let
-  version = "9.0.0";
+  version = "10.0.0";
+
+  archFlags =
+    if stdenv.targetPlatform.isAarch32
+    then [ "--disable-lib32" "--disable-lib64" "--enable-libarm32" ]
+    else if stdenv.targetPlatform.isAarch64
+    then [ "--disable-lib32" "--disable-lib64" "--enable-libarm64" ]
+    else if stdenv.targetPlatform.isx86_32
+    then [ "--enable-lib32" "--disable-lib64" ]
+    else if stdenv.targetPlatform.isx86_64
+    then [ "--disable-lib32" "--enable-lib64" ]
+    else [];
```

Comment:
I would use `lib.enableFeature` for these to avoid O(n^2) string literals.

Reply:
Done.